parse.go

```diff
@@ -694,20 +748,39 @@ func isFlag(s string) bool {
 	return strings.HasPrefix(s, "-") && strings.TrimLeft(s, "-") != ""
 }
 
-// val returns a reflect.Value corresponding to the current value for the
-// given path
-func (p *Parser) val(dest path) reflect.Value {
+// getVal returns a reflect.Value corresponding to the current value for
+// the given path, but does not instantiate nil values in the path.
+func (p *Parser) getVal(dest path) reflect.Value {
 	v := p.roots[dest.root]
 	for _, field := range dest.fields {
 		if v.Kind() == reflect.Ptr {
 			if v.IsNil() {
-				return reflect.Value{}
+				v = reflect.New(v.Type().Elem())
```

Comment:
When would you run into this case? If someone calls `getVal` and the value corresponding to the path doesn't exist because there is a nil pointer on the path then it's going to be a little confusing to debug since you'll end up reading a value from some object that is not actually in the command struct. This will be doubly confusing if this function returns a struct that then gets filled in with argument values because when debugging it will look as if the struct is being filled in, but then at the end of it all the struct will be absent from the top-level command struct. Perhaps `panic` in this case and set up the code that calls this so that it only never gets called for uninitialized paths.

Reply:
Indeed. This function was only called for 1 use-case. That logic has been folded into the function to simplify the code where it was called from, and to avoid this confusion.

In all other cases the original `val(...)` function can be used still, which is now explicitly nil-pointer safe.